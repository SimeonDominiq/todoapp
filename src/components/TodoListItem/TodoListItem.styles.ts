import styled from "styled-components"

const TodoListItemWrapper = styled.div`
  margin-bottom: 5px;
  label {
    cursor: pointer;
  }
`
export default TodoListItemWrapper
