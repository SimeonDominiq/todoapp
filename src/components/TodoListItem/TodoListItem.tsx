import { FC } from "react"
import { useDispatch } from "react-redux"
import { updateStatus } from "src/store/todo/todoSlice"
import { AppDispatch } from "../../store/store"
import TodoListItemWrapper from "./TodoListItem.styles"

type Props = {
  groupId: number
  id: number
  title: string
  isCompleted: boolean
}

const TodoListItem: FC<Props> = ({ groupId, id, title, isCompleted }) => {
  const dispatch = useDispatch<AppDispatch>()
  return (
    <TodoListItemWrapper>
      <label>
        <input
          type="checkbox"
          value={id}
          onChange={() => {
            dispatch(updateStatus({ isCompleted: !isCompleted, groupId, id }))
          }}
          checked={isCompleted}
        />{" "}
        {title}
      </label>
    </TodoListItemWrapper>
  )
}

export default TodoListItem
