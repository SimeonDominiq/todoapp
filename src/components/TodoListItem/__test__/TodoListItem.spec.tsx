import renderer from "react-test-renderer"
import { Provider } from "react-redux"
import configureStore from "redux-mock-store"
import TodoListItem from "../TodoListItem"

const todoItem = {
  groupId: 1,
  id: 11,
  title: "Setup virtual office",
  isCompleted: false,
}

const mockStore = configureStore()

describe("App: Components: Todo List Item Layout", () => {
  let wrapper: any

  it("defines the component", () => {
    wrapper = renderer.create(
      <Provider store={mockStore()}>
        <TodoListItem {...todoItem} />
      </Provider>,
    )

    expect(wrapper).toBeDefined()
  })

  it("matches snapshot", () => {
    const output = wrapper.toJSON()
    expect(output).toMatchSnapshot()
  })
})
