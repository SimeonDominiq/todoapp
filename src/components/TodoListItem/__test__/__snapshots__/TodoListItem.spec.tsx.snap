// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App: Components: Todo List Item Layout matches snapshot 1`] = `
<div
  className="sc-bczRLJ kLtGVI"
>
  <label>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
      value={11}
    />
     
    Setup virtual office
  </label>
</div>
`;
